### PR TITLE
Move NonNegDuration from tmp-api and update ProposalClasses

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/model/package.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+package lucuma.core
+
+import cats.Eq
+import cats.Monoid
+import cats.syntax.all._
+import eu.timepit.refined.api.{ Refined, RefinedTypeOps, Validate }
+import eu.timepit.refined.api.Validate.Plain
+import eu.timepit.refined.boolean.Not
+import eu.timepit.refined.numeric.{ GreaterEqual, Interval, Less, NonNegative }
+import shapeless.Nat._0
```
Comment (cquiroz) [inline modules/core/shared/src/main/scala/lucuma/core/model/package.scala:13]: Can this be done without shapeless?

**Review by cquiroz — APPROVED** [review]
LGTM

File: modules/core/shared/src/main/scala/lucuma/core/model/package.scala (hunk 1 above)
Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/model/package.scala:13]: I really don't know. I just copied it from the api. 😄 I can look into it a little. Maybe @swalker2m knows more?

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/model/package.scala:13]: I'll try replacing it.

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/model/package.scala:13]: I'm going to merge, and then see if I can fix this. I'd like to get my api PR in before there are more merge conflicts.

Comment (swalker2m) [reply modules/core/shared/src/main/scala/lucuma/core/model/package.scala:13]: > I'm going to merge, and then see if I can fix this. I'd like to get my api PR in before there are more merge conflicts.

:+1:

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/model/package.scala:13]: It turns out this probably isn't easily possible. It is relying on the fact that `Not` and `Less` are case classes with companion objects with the implicit validation functions. `NonNegative` is a type alias and so cannot have a companion object. I think in order to try to do the validation ourselves, we'd end up using the `shapeless.Nat._0`. Or we'd have to implement the `Nat`s.

Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/model/package.scala:13]: Never mind, I was mostly asking in the context of the scala 3 port

**Pull request merged**